src/main/scala/abalone/Pos.scala

```diff
@@ -235,25 +237,26 @@ case class Pos private (index: Int) extends AnyVal {
     )
   def piotrStr = piotr.toString
 
-  def key               = file.toString + rank.toString
-  override def toString = key
-
+  def key                 = file.toString + rank.toString
+  def officialNotationKey = File(rank.index).getOrElse("").toString + "" + Rank(file.index).getOrElse("").toString
```

Comment:
how is this different to what you removed above?

Reply:
I probably should keep the toString

Reply:
update: so as you know, in official coordinates system, letters and numbers are swaped.
So in order to read positions and moves in a more convenient way I updated the toString to reflect that.

Reply:
Sorry I forgot, in that case what you are doing makes more sense! This works for "square based" boards, which is fine given the current shape etc.
We need to be careful between converting uci and officialNotation, e.g. for testing and using pos. There might be places in lila that use the .toString method on a pos to get the uci notation, which the abalone code is no longer doing.
I'm not sure what is best but perhaps a decision can be made when changing lila.

Reply:
This might be a good place to learn about 's' strings in scala
I would write this as:
`s"${File(rank.index).getOrElse("").toString}${Rank(file.index).getOrElse("").toString}`
You might even be able to remove the `toString` bit
But in this case I'm not too against the string concatenation method you are using, although surely you dont need the `+ "" + ` bit?